nova/virt/vmwareapi/images.py

```diff
@@ -149,9 +149,10 @@ def from_image(cls, context, image_id, image_meta):
             'linked_clone': linked_clone,
             'container_format': container_format,
             'vsphere_location': get_vsphere_location(context, image_id),
-            'owner': image_meta.owner
         }
 
+        if image_meta.obj_attr_is_set('owner'):
+            props['owner'] = image_meta.owner
```

Comment:
Can we get into any of those places still afterwards? Because we need the owner information at that point:
```
vmops.py:                                     project_id=vi.ii.owner,
vmops.py:                                        project_id=vi.ii.owner,
vmops.py:                                        project_id=vi.ii.owner,
vmops.py:                                                     project_id=vi.ii.owner,
vmops.py:                                        vi.ii.owner, 'Images')

```

Reply:
The image owner is only of interest for the import of the images or the creation of images. That are those places, where the attribute is set somehow.

In the case of the migration, I need a `VirtualMachineInstanceConfigInfo`, which creates a `VMwareImage`instance which is then mostly needed for getting the extra-specs. The image-owner doesn't get populated, hence an exception.

The owner attribute should be set otherwise.

Reply:
Can we provide the owner via the project ID of the instance or would that be too much effort? Just so the `VMwareImage` structure doesn't have one more optional attribute that we might rely on in the wrong moment?

Reply:
I rather have it fail, then relying on a wrong attribute.
The owner of the image depends on the context of who creates the `VMwareImage`. 

- Is it created during instance creation? The owner comes from the original image. 
- Is I created for creating an image of an instance? It comes from context of the api-caller.

In the context of the VM-migration, I do not read that attribute, and I am in neither context of the image live-cycle.
So, yes, I can fill it with a random value (and the project-id would essentially be that), in order to avoid having an optional value. But then someone could rely on that value, instead of getting an error.

Reply:
You're right. The owner was previously at some point the project id of the instance, which meant duplication for every project. We switched that to be the actual owner of the image and thus not have duplication of the template VMs anymore. My proposal thus didn't make sense.